### DISCUSSION TITLE
Welcome to AI-For-Beginners Discussions!

### DISCUSSION THREAD

**Discussion by shwars** [microsoft/AI-For-Beginners#103] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other people who are learning AI using our Curriculum. Feel free to ask questions about the content, as well as about using the materials in classroom.

**Comment by Sazameth** [comment]
Hi there all,

I am Sammy. I had a question if this course provides a certificate of completion.
Thanks

**Reply by WirelessLife** [reply]
This course does not provide a certificate of completion but was created as open source so that professors at universities around the world could adopt the lessons into their curriculum.

**Comment by Painloki** [comment]
I'm new to GitHub and i found this course. Idk how to view this course and how to learn. Github is confusing 🙂. Please help me.

**Reply by WirelessLife** [reply]
Here is a quick start on GitHub use that might help: https://docs.github.com/en/get-started/quickstart/hello-world

**Reply by Mikomijie** [reply]
Hey  man how's it going now , still commited to the ai journey?

**Comment by Painloki** [comment]
Thank you sir.

**Comment by Efonde** [comment]
Hello  
good day , i don,t understand how the  assignment is to be done ,like are we just to do it and keep or submit

**Reply by WirelessLife** [reply]
The assignments can be completed by yourself for self-skilling. They do not need to be submitted.

**Comment by Efonde** [comment]
Alright
Thank you

**Comment by Efonde** [comment]
My question is how do I know that what I do is correct like if I'm on track

**Reply by WirelessLife** [reply]
So the AI-For-Beginners content is built as open-source curriculum for everyone from professors to students to incorporate into their everyday learnings / teachings.  Answers should be provided in the lesson plans created for educators.

**Comment by Efonde** [comment]
Thank you!

**Comment by bobbyreed** [comment]
Thank you! Great structure! :heart:

**Comment by BadrMazro** [comment]
Hi everyone!
I have tried to run the environment in my local device following these instructions: 
https://github.com/microsoft/AI-For-Beginners/blob/main/etc/how-to-run.md
but at the end I run into dependency issue says: 
*ERROR: Cannot install -r /Users/bader/ai-for-beginners/.devcontainer/requirements.txt (line 22) and tokenizers==0.10.3 because these package versions have conflicting dependencies.*
I even tried to run it on Binder cloud jubyter also shows the same conflicting dependencies issue
I would appreciate any help :)

**Reply by GanGral** [reply]
Have the same issue:
The conflict is caused by:
    The user requested tokenizers==0.10.3
    transformers 4.30.0 depends on tokenizers!=0.11.3, <0.14 and >=0.11.1

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

\
Pip subprocess error:
ERROR: Cannot install -r \Projects\ai-for-beginners\.devcontainer\requirements.txt (line 22) and tokenizers==0.10.3 because these package versions have conflicting dependencies.
ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/user_guide/#fixing-conflicting-dependencies

**Reply by GanGral** [reply]
fixed by changing tokenizers version in requirements.txt:

tokenizers==0.13.3

**Reply by lufonzi** [reply]
Changing the tokenizers version did work for me, thank you @GanGral

**Comment by lufonzi** [comment]
I am NOT a developer and I am trying to follow the AI for Beginners course. I have done everything possible to follow the instructions to execute the code included in the course, but no matter how hard I try, I cannot go beyond visualizing the code. I am lost and do not know what to expect when I see that a topic comes with a Notebook or a PyTorch or a Tensorflow or a Lab. I have gone through the contents explained in the "How to Run the Code" page, I have gone through the post from Dmitri Soshnikov but nothing has enlightened me... Any suggestion for me to be able to progress?
Thank you

**Comment by mosoksehbro** [comment]
Its important

**Comment by ghost** [comment]
Muito importante para o bom entendimento de todos e sem falar que ideias novas podem surgir e serem discutidas.
E para aqueles que tiver dúvidas tem onde consultar.

**Comment by STop211650** [comment]
When I run 
conda env create --name ai4beg --file .devcontainer/environment.yml

I get these dependency errors. Similar problem to @GanGral but different errors. Does anyone have an updated requirements.txt file that works for them? I am running an M2 MacBook Air.

Solving environment: failed

PackagesNotFoundError: The following packages are not available from current channels:

  - conda-forge::opencv==4.5
  - pytorch::pytorch==1.11.0
  - pytorch::torchdata==0.3.0
  - pytorch::torchtext==0.12.0
  - pytorch::torchvision==0.12.0

**Reply by wilkersoh** [reply]
@STop211650 can try to change those complain version listed in your terminal.

below list version working fine to me

```
name: ai4beg
channels:
  - defaults
dependencies:
  - ipykernel=6.4.1
  - ipython=7.27.0
  - ipywidgets=7.6.5
  - jupyter=1.0.0
  - matplotlib=3.4.2
  - numpy=1.22.3
  - python=3.8.12
  - requests=2.26.0
  - scikit-learn=0.24.2
  - scipy=1.6.2
  - pip=21.0.1
  - conda-forge::opencv=4.5
  - setuptools=58.0.4
  - pytorch::pytorch=1.12.1
  - pytorch::torchtext=0.13.1
  - pytorch::torchvision=0.13.1
  - pytorch::torchdata=0.4.1
  - pip:
      - -r requirements.txt

``

**Comment by SuchanTso** [comment]
Hi , I'm new here learning AI.
I'm try to complete all Labs here, and I've already made it to product some.
How I suppose to know the right solutions that may help for me to check if I'm right.
Thx

**Reply by mosoksehbro** [reply]
Try looking for references regarding Ai Model Cycle and learning path Machine learning

**Comment by Jesseekoh** [comment]
Hi, I'm new to the course and I want to ask if it's a must to take the Ai for beginners course before taking the machine learning for beginners course?

**Comment by DK-KD-DK** [comment]
Open-source initiatives like "AI for Beginners" are invaluable for making AI learning accessible to everyone, regardless of their background. It's great to see discussions addressing practical issues like dependencies and implementation challenges.

Exciting news for beginners: Klover.ai will soon release a comprehensive open-source catalog packed with beginner-friendly examples, categorized by use case. This resource aims to simplify learning and provide hands-on guidance for common AI workflows. Stay tuned for updates!

A couple of thoughts/questions:

1)  Would a centralized troubleshooting guide help streamline resolving common dependency errors for users?
2)  How could frameworks like Klover.ai and their Artificial General Decision Making™ (AGD™) inspire new AI projects, especially for real-world problem-solving?

Looking forward to seeing how the community evolves with resources like these!

**Comment by ItsmeKY** [comment]
Hi, I am new to this course, I am trying to access the pre-post lecture exams but the link gives 404 error. What to do?

**Reply by zach-exe** [reply]
did you ever figure out a solution? I'm getting a 404 error for the quizzes.

**Comment by Mikomijie** [comment]
Hey , I'm just getting started with this course and  its looking deep, for those who have gone far can you give me some advice

**Comment by ellielo** [comment]
Even as machines reshape how work gets done, imagination and choice still come from people. Routines may shift to software, yet direction and values stay rooted in humans. When crafting material around artificial intelligence, mixing smart systems with personal touch tends to work well. Platforms such as [Clever AI Humaniser](https://cleverhumanizer.ai/ai-writer)  ease the process, turning robotic text into something smoother and livelier. Few small gaps exist - tone choice isn’t ready - but still works well without cost. What comes next leans on teamwork, not just machines.

**Comment by qa4examofficial** [comment]
Master **Microsoft AI-103  certification** with comprehensive **practice questions, study guides,** and mock tests. Download free demo and prepare effectively with updated materials.
https://www.qa4exam.com/microsoft/ai-103-free

**Comment by bubamuhammad223-boop** [comment]
holle___

**Comment by 23HughAI** [comment]
Good discussion

[a comment by an automated account is not shown]
